extensions/amp-analytics/0.1/variables.js

```diff
@@ -0,0 +1,263 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {cryptoFor} from '../../../src/crypto';
+import {dev, user} from '../../../src/log';
+import {fromClass} from '../../../src/service';
+import {isArray, isFiniteNumber, map} from '../../../src/types';
+
+/** @const {String} */
+const TAG = 'Analytics.Variables';
+
+/** @const {RegExp} */
+const VARIABLE_ARGS_REGEXP = /([^(]*)(\([^)]*\))?/;
+
+/** @typedef {function(...?):(?|!Promise<?>)} */
+let VariableFilterDef;
+
+/**
+ * @param {!string} str
```

Comment:
So, in truth, these are always strings, right? If yes - let's declare them as strings; type conversions are already explicit in the code.